### PR TITLE
Migrates from our custom ListenableBuilder to Flutter's new ListenableBuilder (Resolves #1132)

### LINKED ISSUE #1132: Migrate from our custom ListenableBuilder to Flutter's new ListenableBuilder
We defined a widget called `ListenableBuilder` so that we could rebuild when a `Listenable` changed. This could always be done with an `AnimatedBuilder`, but that name was confusing, because we weren't watching animations.

Flutter recently made a similar change in the framework. As a result, there's a naming conflict in `super_editor`. Up to this point, we've explicitly hidden Flutter's own `ListenableBuilder` in favor of our own. 

For this ticket, get rid of our implementation of `ListenableBuilder` in favor of Flutter's implementation.

### PR DESCRIPTION
**Pull request by rutvik110**
Removed our custom `ListenableBuilder` in favour of newly introduced `ListenableBuilder` in Flutter. (Resolves #1132)

### CODE REVIEW

File: super_editor/lib/super_editor.dart [hunk 1]
```diff
@@ -24,12 +23,10 @@ export 'src/default_editor/default_document_editor.dart';
 export 'src/default_editor/default_document_editor_reactions.dart';
 export 'src/default_editor/document_caret_overlay.dart';
 export 'src/default_editor/document_focus_and_selection_policies.dart';
-export 'src/infrastructure/document_gestures.dart';
 export 'src/default_editor/document_gestures_mouse.dart';
 export 'src/default_editor/document_gestures_touch.dart';
-export 'src/infrastructure/document_gestures_interaction_overrides.dart';
-export 'src/default_editor/document_ime/document_input_ime.dart';
 export 'src/default_editor/document_hardware_keyboard/document_input_keyboard.dart';
+export 'src/default_editor/document_ime/document_input_ime.dart';
```
Comment (matthew-carroll) [inline super_editor/lib/super_editor.dart:29]: It looks like there are a number of changes in this file. Seems to me that the only change should be the removal of the `ListenableBuilder` export.

File: super_editor/lib/src/infrastructure/super_textfield/ios/_editing_controls.dart [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import 'dart:math';
 
 import 'package:flutter/material.dart';
-import 'package:super_editor/src/infrastructure/_listenable_builder.dart';
 import 'package:super_editor/src/infrastructure/_logging.dart';
-import 'package:super_editor/src/infrastructure/platforms/mobile_documents.dart';
-import 'package:super_editor/src/infrastructure/toolbar_position_delegate.dart';
+import 'package:super_editor/src/infrastructure/multi_listenable_builder.dart';
 import 'package:super_editor/src/infrastructure/platforms/ios/magnifier.dart';
+import 'package:super_editor/src/infrastructure/platforms/mobile_documents.dart';
```
Comment (matthew-carroll) [inline super_editor/lib/src/infrastructure/super_textfield/ios/_editing_controls.dart:7]: Why are there are non-listenable builder changes here?

File: super_editor/lib/src/infrastructure/super_textfield/android/_editing_controls.dart [hunk 3]
```diff
@@ -2,18 +2,17 @@ import 'dart:async';
...
 
 import 'package:flutter/material.dart';
-import 'package:super_editor/src/infrastructure/_listenable_builder.dart';
 import 'package:super_editor/src/infrastructure/_logging.dart';
+import 'package:super_editor/src/infrastructure/multi_listenable_builder.dart';
 import 'package:super_editor/src/infrastructure/platforms/android/magnifier.dart';
 import 'package:super_editor/src/infrastructure/super_textfield/android/android_textfield.dart';
 import 'package:super_editor/src/infrastructure/super_textfield/infrastructure/attributed_text_editing_controller.dart';
 import 'package:super_editor/src/infrastructure/super_textfield/infrastructure/text_scrollview.dart';
+import 'package:super_editor/src/infrastructure/super_textfield/metrics.dart';
```
Comment (matthew-carroll) [inline super_editor/lib/src/infrastructure/super_textfield/android/_editing_controls.dart:11]: Why was this moved?

Comment (rutvik110) [reply super_editor/lib/src/infrastructure/super_textfield/android/_editing_controls.dart]: Had an IDE setting on to organize imports that resulted in all this 3 issues you pointed out. Just reverted those and pushed the latest updates.

Btw, is there a specific reasons as to why some files name starts with "_"? What does it imply?

Comment (matthew-carroll) [reply super_editor/lib/src/infrastructure/super_textfield/android/_editing_controls.dart]: Initially I used "_" to indicate the intention for private files, but it proved pointless. Eventually we'll remove all the underscores.

**Review by matthew-carroll — APPROVED** [review]
LGTM

**Comment by matthew-carroll** [comment]
@rutvik110 I merged this PR into `main`. We also maintain a `stable` branch which tracks Flutter's `stable` branch. Therefore, every time we merge a PR into `main` that also works for Flutter's `stable`, we need to cherry pick the merged commit to a branch off of `stable`, and then merge that to `stable`.

Please create a branch off of `stable` called something like `cherry-pick-to-stable` and then `git cherry-pick` the merged commit from `main` to that branch, and open a PR to merge that branch into `stable`. 

This is something that you should do automatically after we merge any one of your PRs into `main`.

**Comment by rutvik110** [comment]
Sure! On it👍
